integration_tests/src/bin/benchmark/benchmark_pass.rs

```diff
@@ -0,0 +1,340 @@
+                output_options: EncodedDataOutputOptions {
+                    audio: None,
+                    video: Some(VideoEncoderOptions::H264(ffmpeg_h264::Options {
+                        preset,
+                        resolution: compositor_render::Resolution {
+                            width: self.output_resolution.width,
+                            height: self.output_resolution.height,
+                        },
+                        raw_options: Vec::new(),
+                    })),
+                },
+            },
+        )?;
+        Ok(Box::new(result))
+    }
+
+    fn register_pipeline_raw_output(
+        &self,
+        pipeline: &Arc<Mutex<Pipeline>>,
+        output_id: &OutputId,
+        root: Component,
+    ) -> Result<Box<dyn DurationReceiver + Send>, RegisterOutputError> {
+        let result = Pipeline::register_raw_data_output(
+            pipeline,
+            output_id.clone(),
+            RegisterOutputOptions {
+                video: Some(OutputVideoOptions {
+                    initial: root,
+                    end_condition: PipelineOutputEndCondition::Never,
+                }),
+                audio: None,
+                output_options: RawDataOutputOptions {
+                    audio: None,
+                    video: Some(RawVideoOptions {
+                        resolution: compositor_render::Resolution {
+                            width: self.output_resolution.width,
+                            height: self.output_resolution.height,
+                        },
+                    }),
+                },
+            },
+        )?;
+        Ok(Box::new(result.video.unwrap()))
+    }
+
+    fn register_pipeline_mp4_input(
+        &self,
+        pipeline: &Arc<Mutex<Pipeline>>,
+        input_id: &InputId,
+        path: &Path,
+    ) -> Result<InputInitInfo, RegisterInputError> {
+        let video_decoder = self.video_decoder;
+        Pipeline::register_input(
+            pipeline,
+            input_id.clone(),
+            RegisterInputOptions {
+                input_options: InputOptions::Mp4(Mp4Options {
+                    should_loop: true,
+                    video_decoder,
+                    source: Source::File(path.to_path_buf()),
+                }),
+                queue_options: QueueInputOptions {
+                    offset: None,
+                    required: true,
+                    buffer_duration: None,
+                },
+            },
+        )
+    }
+
+    fn register_pipeline_raw_input(
+        &self,
+        pipeline: &Arc<Mutex<Pipeline>>,
+        input_id: InputId,
+    ) -> Result<Sender<PipelineEvent<Frame>>, RegisterInputError> {
+        let input = Pipeline::register_raw_data_input(
+            pipeline,
+            input_id,
+            RawDataInputOptions {
+                video: true,
+                audio: false,
+            },
+            QueueInputOptions {
+                offset: None,
```

Comment:
offset needs to be resolved based on the first frame, we don't want to have bunch of frames already in buffer when starting